common/src/main/java/org/opensearch/sql/common/utils/StringUtils.java

```diff
@@ -27,16 +27,28 @@ public static String unquote(String text, String mark) {
 
   /**
    * Unquote Identifier which has " or ' or ` as mark.
+   * Replace doubled quotes within the string with single version
    * @param text string
    * @return An unquoted string whose outer pair of (single/double/back-tick) quotes have been
    *     removed
    */
   public static String unquoteText(String text) {
-    if (isQuoted(text, "\"") || isQuoted(text, "'") || isQuoted(text, "`")) {
-      return text.substring(1, text.length() - 1);
-    } else {
+
```

Comment:
A few things can be improved here (lines 36-50).

Currently, the input string is scanned multiple times (3) and temporary strings (3) are allocated in each scenario. 

A more efficient implementation is possible by scanning the input string once and using `StringBuilder` to allocate only one string. In pseudocode it is something like:
```
IF input_string IsQuoteBy $QUOTE
FOREACH char in input_string[1, LENGTH - 1):
   IF char is '\' ADD next char to StringBuilder
   ELSE ADD char to StringBuilder
       
```
I don't think we even need to check which quote character the string is surrounded by.

See this StackOverflow [post](https://stackoverflow.com/a/552945) and this [method](https://github.com/antlr/antlr4/blob/9abda183db8e241cb216308652bf76d2938d1f47/tool/src/org/antlr/v4/misc/CharSupport.java#L86) for reference.

This would also avoid the repeated code pattern -- `if quoted by $QUOTE then ... `.

Reply:
@MaxKsyunz  Can you please review my updated code for StringUtils.java? It does not use `StringBuilder`, but it resolves other issues. If you still recommend I use `StringBuilder`, I will figure it out and use it (thanks for the documentation).

Reply:
If you want to reduce the number of temporary strings, we should consider using match replaceAll with a pattern.  https://www.geeksforgeeks.org/matcher-replaceallstring-method-in-java-with-examples/

If we only want a single isQuoted call (which returns the quoted character) we could reduce this from 3 IF statements, to 2.  But the backticks is a unique case that doesn't reduce double-backticks to a single backtick.